### PR TITLE
Updated java toolchain

### PR DESCRIPTION
**Pull request by fbarbari**
This PR references [this issue](https://github.com/compiler-explorer/compiler-explorer/issues/6289) and [this PR](https://github.com/compiler-explorer/infra/pull/1272).

New Java versions (and corresponding groups):
 - 8u402b06
 - 9.0.4
 - 10.0.2
 - 22.0.0

Updated Java versions:
 - 12.0.1 -> 12.0.2
 - 17.0.0 -> 17.0.2
 - 18.0.0 -> 18.0.2
 - 20.0.0 -> 20.0.2
 - 21.0.0 -> 21.0.2

Also updated the default java version to 21.0.2

### CODE REVIEW

File: etc/config/java.amazon.properties [hunk 1]
```diff
@@ -15,19 +15,40 @@ baseName=jdk
...
+
 group.java11.compilers=java1102
 group.java11.groupName=OpenJDK 11
 group.java11.objdumper=/opt/compiler-explorer/jdk-11.0.2/bin/javap
 compiler.java1102.exe=/opt/compiler-explorer/jdk-11.0.2/bin/javac
 compiler.java1102.runtime=/opt/compiler-explorer/jdk-11.0.2/bin/java
 compiler.java1102.semver=11.0.2
 
-group.java12.compilers=java1201
+group.java12.compilers=java1202
```
Comment (junlarsen) [inline etc/config/java.amazon.properties:46]: I believe you should add 1202 in additon to 1201 here (and consequently for the other versions too). Any shortlinks using 1201 will break if we remove the old compiler ids.

**Review by junlarsen — CHANGES_REQUESTED** [review]
Thanks for updating them! I believe the best thing to do here is to add the new compilers in addition to the old ones, otherwise we break backwards compatability

**Review by junlarsen — APPROVED** [review]
This looks good now, thanks a lot!

**Comment by partouf** [comment]
this is now live

**Pull request merged**